articles/getting-started-with-ui-layouts-in-android/index.md

```diff
@@ -0,0 +1,140 @@
+
+**Note: An error will always be displayed if the views are not constrained.**
+#### Attributes
+- `android:id` - This is a unique ID signed to a specific layout.
+- `layout_constraintTop_toTopOf` , `layout_constraintTop_toBottomOf`,`layout_constraintBottom_toTopOf`,`layout_constraintBottom_toBottomOf` - Allign the bottom or top of a desired view to the top or bottom of another view.
+- `layout_constraintLeft_toTopOf` ,`layout_constraintLeft_toBottomOf` ,`layout_constraintLeft_toLeftOf` ,`layout_constraintLeft_toRightOf`,`layout_constraintRight_toTopOf` ,`layout_constraintRight_toBottomOf` ,`layout_constraintRight_toLeftOf`,`layout_constraintRight_toRightOf`  - Align the right or left of a desired view to the right or left of another view.
+- `layout_constraintVertical_bias`, `layout_constraintHorizontal_bias` - Used to position a view either vertically or horizontally at a particular point.
+
+#### Pros and Cons
+#### Pros
+- It is easy to use because of its drag and drop feature.
+- It is very flexible.
+- It can supplant various layouts that are Relative, Frame, and  Linear.
+- It is straightforward to create a flat view hierarchy.
+- Constraint layout has an improved performance over other layouts.
+- Can support animations with a few lines of code.
+
+### Cons
+- It is quite hard to master, especially for a beginner developer.
+- It can be complex to use when the UI is very basic.
+
+### D) Frame Layout
+The Frame layout is one of the most basic and most straightforward layouts. This layout blocks out an area on the screen to display a single view group. The layout uses the `<FrameLayout>` tag. It is suitable to use this layout when the UI of an application only requires one view. One can use it either as a root element or a view group.
+
+![Frame Layout](/engineering-education/getting-started-with-android-ui-layouts/frame.jpg)
+
+#### Attributes
+- `android:id` - This is a unique ID signed to a specific layout.
+- `android:foreground` - Refers to the drawable to be displayed over the content.
+- `android:foregroundGravity` - Defines the gravity direction applied to the foreground drawable.
+Examples of the values are `center_vertical`, `center_horizontal`, top, bottom, left, right, and many others.
+
+#### Pros and Cons
+#### Pros
+- It is very easy to use and understand.
+- It is suitable for a basic UI.
+
+#### Cons
+- It is challenging to use it for a complex UI.
+
+### View group layouts used in Android
+The following are layouts popularly used as view groups:
+### A) ScrollView
+A `ScrollView` is a view group that allows scrollable content. Scrolling in the `ScrollView` is either vertically or horizontally. The layout uses the `<ScrollView>` tag. A `ScrollView` can only support one direct child; otherwise, one must add view groups, e.g., Linear Layout, if they want the scroll view to be more detailed. One can use it as a view group only.
```

Comment:
The last sentence is not accurate. It can also be used as a root layout. It is just not recommended for that use case. (I stand to be corrected)

Reply:
@kayere Thank you for your suggestion. I thought phrasing the statement as  "most suitable" is still the same as saying what you have stated above. But let me change it so that it is more clear.